internal/project/service.go

```diff
@@ -619,7 +619,9 @@ func (s *Service) getDefaultProjectForScript(scriptInfo *ScriptInfo) *Project {
 
 func (s *Service) createInferredProject(currentDirectory string, projectRootPath tspath.Path) *Project {
 	// !!!
-	compilerOptions := core.CompilerOptions{}
+	compilerOptions := core.CompilerOptions{
+		AllowJs: core.TSTrue,
```

Comment:
We normally set this to true when there are js root files. i got this working in my branch as its kind of what i do for type Acq defaults too.

Reply:
It appears to be unconditional https://github.com/microsoft/TypeScript/blob/d88d3a46810bfae0b072beb023a2b09b8026b82c/src/server/project.ts#L2422

Reply:
You are right. thats the maxNodeModule depth part thats conditional